registry/handlers/hooks.go

```diff
@@ -0,0 +1,103 @@
+package handlers
+
+import (
+	"fmt"
+	"net/smtp"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/distribution/configuration"
+)
+
+// ReportHook is for hooking Panic in web application
+type ReportHook struct {
+}
+
+// Fire forwards an error to ReportHook
+func (hook *ReportHook) Fire(entry *logrus.Entry) error {
+	logrus.Info("Message: ", entry.Message)
+	if errConfig.Log.Handlers.Errors.Type == "email" {
+		if err := mailSending(errConfig, entry.Message); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Levels contains hook levels to be catched
+func (hook *ReportHook) Levels() []logrus.Level {
+	return []logrus.Level{
+		logrus.DebugLevel,
```

Comment:
This should only return the Level for which the hook is configured. Let's say we have this configuration:

```yaml
levels:
  - error
  - panic
```

This method would return something like this:

```
return hooks.Levels
```

Which should be setup with the configuration.